pilot/pkg/proxy/envoy/v1/resources.go

```diff
@@ -705,12 +705,17 @@ type Listeners []*Listener
 // Normalize sorts and de-duplicates listeners by address
 func (listeners Listeners) normalize() Listeners {
 	out := make(Listeners, 0, len(listeners))
-	set := make(map[string]bool)
+	set := make(map[string]*Listener)
 	for _, listener := range listeners {
-		if !set[listener.Address] {
-			set[listener.Address] = true
-			out = append(out, listener)
+		if l, collision := set[listener.Address]; collision {
+			ol, _ := json.Marshal(*l)
+			ll, _ := json.Marshal(*listener)
+			log.Errorf("Listener collision for %s\n---\n%s\n--- rejected ---\n%s", listener.Address,
+				string(ol), string(ll))
+			continue
```

Comment:
This bug is there in v2 code as well. but we don't use the normalize listeners anymore since I use a hash map and do this warning inline.